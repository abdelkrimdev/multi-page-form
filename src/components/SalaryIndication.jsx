import React from 'react'
import PropTypes from 'prop-types'
import { Radio } from 'antd'

const SalaryIndication = ({ onChange }) => {
  const radioStyle = {
    display: 'block',
    height: '30px',
    lineHeight: '30px',
  }

  return (
    <Radio.Group name='salaryIndication' onChange={onChange}>
      <Radio style={radioStyle} value='0 - 1000'>0 - 1000</Radio>
      <Radio style={radioStyle} value='1.000 - 2.000'>1.000 - 2.000</Radio>
      <Radio style={radioStyle} value='2.000 - 3.000'>2.000 - 3.000</Radio>
      <Radio style={radioStyle} value='3.000 - 4.000'>3.000 - 4.000</Radio>
      <Radio style={radioStyle} value='Mehr als 4.000'>Mehr als 4.000</Radio>
    </Radio.Group>
  )
}

SalaryIndication.propTypes = {

}

export default SalaryIndication
