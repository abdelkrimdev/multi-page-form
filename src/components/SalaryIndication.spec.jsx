import React from 'react'
import { shallow } from 'enzyme'

import SalaryIndication from './SalaryIndication'

describe('salary indication component', () => {
  it('should render without crashing.', () => {
    const wrapper = shallow(<SalaryIndication />)

    expect(wrapper).toBeDefined()
  })

  it('should render correctly.', () => {
    const wrapper = shallow(<SalaryIndication />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should set the onChange property.', () => {
    const handleChange = jest.fn()
    const wrapper = shallow(<SalaryIndication onChange={handleChange} />)
    
    wrapper.prop('onChange')()

    expect(handleChange).toBeCalled()
  })
})
