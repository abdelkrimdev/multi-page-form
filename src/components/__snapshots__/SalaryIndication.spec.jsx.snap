// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`salary indication component should render correctly. 1`] = `
<RadioGroup
  buttonStyle="outline"
  disabled={false}
  name="salaryIndication"
  prefixCls="ant-radio"
>
  <Radio
    prefixCls="ant-radio"
    style={
      Object {
        "display": "block",
        "height": "30px",
        "lineHeight": "30px",
      }
    }
    type="radio"
    value="0 - 1000"
  >
    0 - 1000
  </Radio>
  <Radio
    prefixCls="ant-radio"
    style={
      Object {
        "display": "block",
        "height": "30px",
        "lineHeight": "30px",
      }
    }
    type="radio"
    value="1.000 - 2.000"
  >
    1.000 - 2.000
  </Radio>
  <Radio
    prefixCls="ant-radio"
    style={
      Object {
        "display": "block",
        "height": "30px",
        "lineHeight": "30px",
      }
    }
    type="radio"
    value="2.000 - 3.000"
  >
    2.000 - 3.000
  </Radio>
  <Radio
    prefixCls="ant-radio"
    style={
      Object {
        "display": "block",
        "height": "30px",
        "lineHeight": "30px",
      }
    }
    type="radio"
    value="3.000 - 4.000"
  >
    3.000 - 4.000
  </Radio>
  <Radio
    prefixCls="ant-radio"
    style={
      Object {
        "display": "block",
        "height": "30px",
        "lineHeight": "30px",
      }
    }
    type="radio"
    value="Mehr als 4.000"
  >
    Mehr als 4.000
  </Radio>
</RadioGroup>
`;
